unity-client/Assets/Scripts/MainScripts/DCL/Components/Animator/DCLAnimator.cs

```diff
@@ -71,33 +71,31 @@ private void OnComponentUpdated(DecentralandEntity e)
 
         private void Initialize()
         {
-            if (entity == null) return;
+            if (entity == null || animComponent != null) return;
 
             //NOTE(Brian): fetch all the AnimationClips in Animation component.
-            if (animComponent == null)
-            {
-                animComponent = transform.parent.GetComponentInChildren<Animation>(true);
+            animComponent = transform.parent.GetComponentInChildren<Animation>(true);
 
-                if (animComponent == null) return;
+            if (animComponent == null) return;
 
-                clipNameToClip.Clear();
-                clipToState.Clear();
-                int layerIndex = 0;
+            clipNameToClip.Clear();
+            clipToState.Clear();
+            int layerIndex = 0;
 
-                animComponent.playAutomatically = true;
-                animComponent.enabled = true;
-                animComponent.Stop(); //NOTE(Brian): When the GLTF is created by GLTFSceneImporter a frame may be elapsed, 
-                                      //putting the component in play state if playAutomatically was true at that point.
+            animComponent.playAutomatically = true;
+            animComponent.enabled = true;
+            animComponent.Stop(); //NOTE(Brian): When the GLTF is created by GLTFSceneImporter a frame may be elapsed,
+            //putting the component in play state if playAutomatically was true at that point.
+            animComponent.clip?.SampleAnimation(animComponent.gameObject, 0);
```

Comment:
I don't get this code. Why using `SampleAnimation` here fixes the bug? 

I understand that this issue is coming from `GLTFSceneImporter` setting `playAutomatically` as true before this component comes along and stop it.

Reply:
From the ticket: `When an entity's GLTF or GLB model includes animations, the first one of these will always be played on a loop. That's the desired behavior if no Animator is preset`

The problem is we dont know in advance if an animation will have an Animator or not so for a few frames the animation is playing and then is stopped (which leaves the object in the middle of the animation). That was ruining stuff like doors and such.

So sampling the first frame of the animation will leave it as it is.

Reply:
Let's look at what's happening behind this bug (at least from my understanding):

- GLTF starts loading
- GLTF loads the animation, and sets `playAutomatically` to `true`
- Animation starts playing in loop
- A few frames pass by
- The `DCLAnimator` component is created
- `Awake()` gets called, effectively "stopping" the animation, but too late

Then, how sampling the animation in the `Awake()` fixes it? In this case, the stop will come too late.

If all of this is true, the fix should be along the lines of:

- GLTF starts loading
- GLTF loads the animation, and sets `playAutomatically` to `false`
- GLTF finishes loading
- Here's where the fix should be applied... we should 
      (A) add a `DCLAnimator` component to ALL gltfs or 
      (B) wait a grace time to see if a `DCLAnimator` gltf exists, if not, we should `Play` the animation.
- The `DCLAnimator` component is created
- `Awake()` gets called, and animation should be effectively stopped when reached this stage